### PR TITLE
added AnswerCorrectness and AnswerFaithfulness auto-eval modules

### PR DESCRIPTION
**Pull request by arnavsinghvi11**
AnswerCorrectness - verifies if predicted answer is correct for question based on gold answer and outputs True/False
AnswerFaithfulness - verifies if predicted answer is based off context/reasoning/rationale for question and outputs True/False

(corrects faulty [PR](https://github.com/stanfordnlp/dspy/pull/173))

### CODE REVIEW

File: dspy/evaluate/auto_evaluation.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import dspy 
+
+class AnswerCorrectnessSignature(dspy.Signature):
+    """Determines if predicted answer matches the gold answer."""
```
Comment (okhat) [inline dspy/evaluate/auto_evaluation.py:4]: This is a strange instruction.

File: dspy/evaluate/auto_evaluation.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    def __init__(self):
+        super().__init__()
+        self.evaluate_correctness = dspy.ChainOfThought(AnswerCorrectnessSignature)
+    
+    def forward(self, question, gold_answer, predicted_answer):
+        return self.evaluate_correctness(question=question, gold_answer=gold_answer, predicted_answer=predicted_answer)
+
+
+class AnswerFaithfulnessSignature(dspy.Signature):
+    """Checks if answer for question is based on rationale."""
```
Comment (okhat) [inline dspy/evaluate/auto_evaluation.py:21]: same.

Comment (okhat) [reply dspy/evaluate/auto_evaluation.py:21]: please fix then merge directly!

**Comment by pepo1275** [comment]
Fixed in commits dba2d59b + ba0757c9. Verified with 15-UC functional sim.

**Pull request merged**